receiver/zipkin/scribe/scribe_receiver.go

```diff
@@ -0,0 +1,170 @@
+func NewReceiver(addr string, port uint16, category string) (receiver.TraceReceiver, error) {
+	r := &scribeReceiver{
+		addr: addr,
+		port: port,
+		collector: &scribeCollector{
+			category:            category,
+			msgDecoder:          base64.StdEncoding.WithPadding('='),
+			tBinProtocolFactory: thrift.NewTBinaryProtocolFactory(true, false),
+		},
+	}
+	return r, nil
+}
+
+func (r *scribeReceiver) StartTraceReception(ctx context.Context, destination receiver.TraceReceiverSink) error {
+	r.Lock()
+	defer r.Unlock()
+
+	if destination == nil {
+		return errors.New("trace reception requires a non-nil destination")
+	}
+
+	err := errAlreadyStarted
+	r.startOnce.Do(func() {
+		err = nil
+		r.collector.traceSink = destination
+		serverSocket, sockErr := thrift.NewTServerSocket(r.addr + ":" + strconv.Itoa(int(r.port)))
+		if sockErr != nil {
+			err = sockErr
+			return
+		}
+
+		sockErr = serverSocket.Open()
+		if sockErr != nil {
+			err = sockErr
+			return
+		}
+
+		r.server = thrift.NewTSimpleServer4(
+			scribe.NewScribeProcessor(r.collector),
+			serverSocket,
+			thrift.NewTFramedTransportFactory(thrift.NewTTransportFactory()),
+			thrift.NewTBinaryProtocolFactory(true, false),
+		)
+
+		// The interface doesn't support reporting anything from the async function.
+		go r.server.Serve()
+	})
+
+	return err
+}
+
+func (r *scribeReceiver) StopTraceReception(ctx context.Context) error {
+	r.Lock()
+	defer r.Unlock()
+
+	var err = errAlreadyStopped
+	r.stopOnce.Do(func() {
+		err = r.server.Stop()
+	})
+	return err
+}
+
+// scribeCollector implements the Thrift interface of a scribe server as supported by zipkin.
+// See https://github.com/openzipkin/zipkin/tree/master/zipkin-collector/scribe
+type scribeCollector struct {
+	category            string
+	msgDecoder          *base64.Encoding
+	tBinProtocolFactory *thrift.TBinaryProtocolFactory
+	traceSink           receiver.TraceReceiverSink
+}
+
+var _ scribe.Scribe = (*scribeCollector)(nil)
+
+// Log is the function that receives the messages sent to the scribe server. It is required
+func (sc *scribeCollector) Log(messages []*scribe.LogEntry) (r scribe.ResultCode, err error) {
+	zSpans := make([]*zipkincore.Span, 0, len(messages))
+	for _, logEntry := range messages {
+		if sc.category != logEntry.Category {
+			// Not the specified category, do nothing
+			continue
+		}
+
+		b, err := sc.msgDecoder.DecodeString(logEntry.Message)
+		if err != nil {
+			return scribe.ResultCode_OK, err
+		}
+
+		r := bytes.NewReader(b)
+		st := thrift.NewStreamTransportR(r)
+		zs := &zipkincore.Span{}
+		if err := zs.Read(sc.tBinProtocolFactory.GetProtocol(st)); err != nil {
+			return scribe.ResultCode_OK, err
+		}
+
+		zSpans = append(zSpans, zs)
+	}
+
+	if len(zSpans) == 0 {
+		return scribe.ResultCode_OK, nil
+	}
+
+	ocBatches, err := tracetranslator.ZipkinV1ThriftBatchToOCProto(zSpans)
+	if err != nil {
+		return scribe.ResultCode_OK, err
```

Comment:
This looks out of place, `scribe.ResultCode_OK` yet we encountered an error?

Reply:
I had the same feeling, the thing is that the interface only defines ResultCode_OK and ResultCode_TRY_LATER, so in this case the ResultCode_OK paired with the error seems the appropriate response.